test/lib/jdk/test/lib/SA/SATestUtils.java

```diff
@@ -77,11 +78,54 @@ public static void skipIfCannotAttach() {
 
     /**
      * Returns true if this platform is expected to require extra privileges (running using sudo).
+     * If we are running as root or developer mode is enabled, then sudo is not needed.
      */
     public static boolean needsPrivileges() {
-        return Platform.isOSX() && !Platform.isRoot();
+        if (!Platform.isOSX()) {
+            return false;
+        }
+        if (Platform.isRoot()) {
+            return false;
+        }
+        return !developerModeEnabled();
     }
 
+    /*
+     * Run "DevToolsSecurity --status" to see if developer mode is enabled.
+     */
+    private static boolean developerModeEnabled() {
+        List<String> cmd = new ArrayList<String>();
+        cmd.add("DevToolsSecurity");
+        cmd.add("--status");
+        ProcessBuilder pb = new ProcessBuilder(cmd);
+        Process p;
+        try {
+            p = pb.start();
+            try {
+                p.waitFor();
+            } catch (InterruptedException e) {
+                throw new RuntimeException("DevToolsSecurity process interrupted", e);
+            }
+
+            String out = new String(p.getInputStream().readAllBytes());
+            String err = new String(p.getErrorStream().readAllBytes());
+            System.out.print("DevToolsSecurity stdout: " + out);
+            if (out.equals("")) System.out.println();
```

Comment:
This looks odd - why not just an unconditional println?

Reply:
If it is not "", then `out` will include a newline at the end of the last line, which means you'd end up with an empty line in the output. I considered the following:

            if (!out.equals("")) {
                System.out.print("DevToolsSecurity stdout: " + out);
            }

But it prints nothing in the output if it is "", and I that's not what I was looking for.